packages/syft/src/syft/client/gateway_client.py

```diff
@@ -55,3 +56,11 @@ def proxy_client_for(
         if email and password:
             res.login(email=email, password=password, **kwargs)
         return res
+
+    @property
+    def domains(self):
+        return self.api.services.network.get_peers_by_type(node_type=NodeType.DOMAIN)
```

Comment:
Can we also add the check for the network service:
if self.api is not None and self.api.has_service("network"):

Reply:
Ah good catch. Actually, we don't need `self.api is not None` as the property checks None and fetches + returns the API for us anyway.